### PR TITLE
Switches usage of FastJSON to Jackson for Handling JSONs

### PR DESCRIPTION
**Pull request by sabieber**
### BREAKING CHANGES:
* APIs that used `JSONObject` or `JSONArray` now use `ObjectNode` or `ArrayNode` instead

Fixes: [SIRI-736](https://scireum.myjetbrains.com/youtrack/issue/SIRI-736)

### CODE REVIEW

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/db/es/ElasticEntity.java [hunk 1]
```diff
@@ -157,17 +162,9 @@ private Set<String> parseMatchedQueries() {
...
-            return 0;
-        }
-
-        innerHits = innerHits.getJSONObject(name);
-        if (innerHits == null) {
-            return 0;
-        }
-
-        return innerHits.getJSONObject("hits").getJSONObject("total").getIntValue("value");
+        return Json.tryGetAt(getSearchHit(), Json.createPointer(INNER_HITS, name, "hits", "total", "value"))
```
Comment (andyHa) [inline src/main/java/sirius/db/es/ElasticEntity.java:165]: couldn't we keep the pointers as constants? these are immutable, right?

**Review by andyHa — APPROVED** [review]

File: src/main/java/sirius/db/es/ElasticQuery.java [hunk 2]
```diff
@@ -866,37 +875,37 @@ private JSONObject buildPayload() {
...
-                collapse.put(KEY_INNER_HITS, collapseByInnerHits);
+                collapse.putPOJO(KEY_INNER_HITS, collapseByInnerHits);
             }
-            payload.put(KEY_COLLAPSE, collapse);
+            payload.set(KEY_COLLAPSE, collapse);
         }
 
         if (suggesters != null && !suggesters.isEmpty()) {
-            payload.put(KEY_SUGGEST, new JSONObject().fluentPutAll(suggesters));
+            payload.set(KEY_SUGGEST, Json.createObject().setAll(suggesters));
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/es/ElasticQuery.java:908]:
```suggestion
            payload.set(KEY_SUGGEST, Json.convertFromMap(suggesters));
```

File: src/main/java/sirius/db/es/ElasticQuery.java [hunk 3]
```diff
@@ -960,10 +969,10 @@ public long count() {
 
         String filteredRouting = checkRouting(Elastic.RoutingAccessMode.READ);
 
-        JSONObject countResponse = client.count(computeEffectiveIndexName(elastic::determineReadAlias),
+        ObjectNode countResponse = client.count(computeEffectiveIndexName(elastic::determineReadAlias),
                                                 filteredRouting,
                                                 buildSimplePayload());
-        return countResponse.getLong(KEY_COUNT);
+        return countResponse.get(KEY_COUNT).asLong();
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/es/ElasticQuery.java:975]: require

File: src/main/java/sirius/db/es/ElasticQuery.java [hunk 4]
```diff
@@ -1302,19 +1310,18 @@ public void tryTruncate() throws OptimisticLockException {
...
         String filteredRouting = checkRouting(Elastic.RoutingAccessMode.WRITE);
-        JSONObject deleteByQueryResponse = elastic.getLowLevelClient()
+        ObjectNode deleteByQueryResponse = elastic.getLowLevelClient()
                                                   .deleteByQuery(computeEffectiveIndexName(elastic::determineWriteAlias),
                                                                  filteredRouting,
                                                                  buildSimplePayload());
-        if (Boolean.TRUE.equals(deleteByQueryResponse.getBoolean(KEY_TIMED_OUT))
-            || deleteByQueryResponse.getIntValue(KEY_VERSION_CONFLICTS) > 0) {
-            Elastic.LOG.WARN("Truncate timed out or had version conflicts:\n" + deleteByQueryResponse.toJSONString());
+        if (Boolean.TRUE.equals(deleteByQueryResponse.get(KEY_TIMED_OUT).asBoolean())
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/es/ElasticQuery.java:1317]: require or path? directly use boolean

File: src/main/java/sirius/db/es/LowLevelClient.java [hunk 5]
```diff
@@ -293,7 +290,7 @@ public boolean checkTaskActivity(String taskId) {
                     }
                 }).response();
 
-        return !response.containsKey("notFound") && !response.getBooleanValue("completed");
+        return !response.has("notFound") && !response.get("completed").asBoolean();
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/es/LowLevelClient.java:293]: require?

File: src/main/java/sirius/db/es/suggest/TermSuggestion.java [hunk 6]
```diff
@@ -30,11 +31,11 @@ public class TermSuggestion {
...
      * @param json a JSON snipped as returned by ES
      */
-    public TermSuggestion(JSONObject json) {
-        this.text = json.getString(PARAM_TEXT);
-        this.score = json.getFloatValue(PARAM_SCORE);
-        this.highlighted = json.getString(PARAM_HIGHLIGHTED);
-        this.collateMatch = json.getBooleanValue(PARAM_COLLATE_MATCH);
+    public TermSuggestion(ObjectNode json) {
+        this.text = Json.tryValueString(json, PARAM_TEXT).orElse(null);
+        this.score = json.get(PARAM_SCORE).floatValue();
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/es/suggest/TermSuggestion.java:36]: require

File: src/main/java/sirius/db/es/suggest/TermSuggestion.java [hunk 7]
```diff
@@ -30,11 +31,11 @@ public class TermSuggestion {
...
+        this.highlighted = Json.tryValueString(json, PARAM_HIGHLIGHTED).orElse(null);
+        this.collateMatch = json.get(PARAM_COLLATE_MATCH).asBoolean();
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/es/suggest/TermSuggestion.java:38]: require

File: src/main/java/sirius/db/es/suggest/TextPartSuggestion.java [hunk 8]
```diff
@@ -33,12 +34,14 @@ public class TextPartSuggestion {
...
-    public TextPartSuggestion(JSONObject json) {
-        this.text = json.getString(PARAM_TEXT);
-        this.offset = json.getIntValue(PARAM_OFFSET);
-        this.length = json.getIntValue(PARAM_LENGTH);
-        this.termSuggestions =
-                json.getJSONArray(PARAM_OPTIONS).stream().map(JSONObject.class::cast).map(TermSuggestion::new).toList();
+    public TextPartSuggestion(ObjectNode json) {
+        this.text = Json.tryValueString(json, PARAM_TEXT).orElse(null);
+        this.offset = json.get(PARAM_OFFSET).asInt();
+        this.length = json.get(PARAM_LENGTH).asInt();
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/es/suggest/TextPartSuggestion.java:40]: require

File: src/main/java/sirius/db/qdrant/Match.java [hunk 9]
```diff
@@ -25,10 +26,10 @@ public class Match {
      *
      * @param match the match as returned by qdrant
      */
-    public Match(JSONObject match) {
-        this.id = match.getString("id");
-        this.score = match.getFloat("score");
-        this.payload = match.getJSONObject("payload");
+    public Match(ObjectNode match) {
+        this.id = Json.tryValueString(match, "id").orElse(null);
+        this.score = match.get("score").floatValue();
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/qdrant/Match.java:31]: require

File: src/main/java/sirius/db/util/Tensors.java [hunk 10]
```diff
@@ -43,10 +43,10 @@ public static float[] fromList(List<Number> vector) {
      * @param vector the JSON array to extract the float array from
      * @return the extracted float array
      */
-    public static float[] fromJSON(JSONArray vector) {
+    public static float[] fromJSON(ArrayNode vector) {
         float[] result = new float[vector.size()];
         for (int i = 0; i < vector.size(); i++) {
-            result[i] = vector.getFloat(i) == null ? 0 : vector.getFloat(i);
+            result[i] = vector.get(i).isFloat() ? vector.get(i).floatValue() : 0;
```
Comment (mkeckmkeck) [inline src/main/java/sirius/db/util/Tensors.java:49]: asFloat returns 0 as default value

**Review by jakobvogel — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**